nms/app/packages/magmalte/app/components/DataGrid.js

```diff
@@ -163,39 +163,49 @@ function DataObscure(value: number | string, category: ?string) {
   );
 }
 
-function DataCollapse(
-  category: ?string,
-  value: string | number,
-  //$FlowFixMe TODO: Needs a ComponentType argument
-  collapse: ComponentType,
-) {
+function DataCollapse(data: Data) {
   const props = {collapsed: true};
   const classes = useStyles(props);
   const [open, setOpen] = React.useState(true);
+  const dataEntryValue = data.value + (data.unit ?? '');
   return (
-    <List key={`${category ?? value}Collapse`} className={classes.list}>
+    <List
+      key={`${data.category ?? data.value}Collapse`}
+      className={classes.list}>
       <ListItem button onClick={() => setOpen(!open)}>
         <CardHeader
-          data-testid={category}
-          title={category}
+          data-testid={data.category}
+          title={data.category}
           className={classes.dataBox}
-          subheader={value}
+          subheader={
+            data.statusCircle === true
+              ? StatusIndicator(
+                  data.statusInactive || false,
+                  data.status || false,
+                  dataEntryValue,
+                )
+              : data.icon
+              ? DataIcon(data.icon, dataEntryValue)
+              : data.obscure === true
+              ? DataObscure(data.value, data.category)
+              : dataEntryValue
```

Comment:
not a fan of this much ternary chaining. Pull it out to explicit code blocks IMO, but up to you overall.